dali/nvimgcodec/nvimgcodec_wrap.cc

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2021-2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cuda.h>
+#include <dlfcn.h>
+#include <nvimgcodec_version.h>
+#include <stdio.h>
+#include <iostream>
+#include <mutex>
+#include <string>
+#include <unordered_map>
+
+#define STR_IMPL_(x) #x
+#define STR(x) STR_IMPL_(x)
+#define FULL_VER_STR               \
+  STR(NVIMGCODEC_EXT_API_VER_MAJOR) \
+  "." STR(NVIMGCODEC_EXT_API_VER_MINOR) "." STR(NVIMGCODEC_EXT_API_VER_PATCH)
+#define MAJOR_VER_STR STR(NVIMGCODEC_EXT_API_VER_MAJOR)
+
+namespace {
+
+typedef void *NVIMGCODECDRIVER;
+
+static const char __NvimgcodecLibNameFullVer[] = "libnvimgcodec.so." FULL_VER_STR;
+static const char __NvimgcodecLibNameMajorVer[] = "libnvimgcodec.so." MAJOR_VER_STR;
+static const char __NvimgcodecLibName[] = "libnvimgcodec.so";
+static const char __NvimgcodecLibDefaultPathFullVer[] =
```

Comment:
1. C++ says that identifiers starting with a double underscore are reserved.
2. No need to mark contents of an anonymous namespace `static` - it's mostly synonymous.
3.

Reply:
Fixed